src/devices/cpu/z180/z180.cpp

```diff
@@ -1989,13 +1918,18 @@ int z180_device::check_interrupts()
 
 		if (m_irq_state[2] != CLEAR_LINE && (m_itc & Z180_ITC_ITE2) == Z180_ITC_ITE2)
 			m_int_pending[Z180_INT_IRQ2] = 1;
+
+		m_int_pending[Z180_INT_ASCI0] = m_asci_0->check_interrupt();
+		m_int_pending[Z180_INT_ASCI1] = m_asci_1->check_interrupt();
 	}
 
 	for (i = 0; i <= Z180_INT_MAX; i++)
 		if (m_int_pending[i])
 		{
 			cycles += take_interrupt(i);
 			m_int_pending[i] = 0;
+			if (i==Z180_INT_ASCI0) m_asci_0->clear_interrupt();
+			if (i==Z180_INT_ASCI1) m_asci_1->clear_interrupt();
```

Comment:
Do not have OCD so can't tell

Reply:
Nor any cares about not adding to the mountain of tech debt and quality-control issues that already exist, apparently.

Reply:
Main thing I do not like about my current ASCI implementation is all these trampolines, mapping, additional handling just because it is separate device. I understood that device_serial_interface is preferred way of doing implementation. But if it is not then it would make more sense to handle it as part of z180 (like the other parts DMA, CTC,...). @MooglyGuy  Whole point of making this PR was to get feedback, also asked that on Discord since I did not contributed for a time, so I do care about tech debt. But to be honest I do not see code-style things (like spaces,  brackets styles ...) and prefer relaying on clang-formatting or similar to make style unique on project. Which to be honest would be hard to do on MAME since it will make all pending changes unusable since it will touch all files for sure if done.

Reply:
`device_serial_interface` is the preferred way of doing high-level emulation of EIA asynchronous serial protocol for black box devices, like the various RS-232 peripherals and high-level emulation of serial keyboards.  For UART-like devices, it often falls over when you need to simulate detailed behaviour.

If these things are really trivial, `device_serial_interface` may be adequate.  But, as the person who wrote `device_buffered_serial_interface` to help with things like the Sun keyboards, given how abstract and high-level it is, I’d question how suitable it is for emulating something like this.

Reply:
Thanks for explanation. Will see how to make this clean.